doc/build/cmake/style.rst

```diff
@@ -0,0 +1,120 @@
+:orphan:
+
+.. _cmake-style:
+
+CMake Style Guidelines
+======================
+
+General Formatting
+------------------
+
+- **Indentation**: Use **2 spaces** for indentation. Avoid tabs to ensure consistency across different environments.
+- **Line Length**: Limit line length to **100 characters** where possible.
+- **Empty Lines**: Use empty lines to separate logically distinct sections within a CMake file.
+- **No Space Before Opening Brackets**: Do not add a space between a command and the opening parenthesis.
+  Use ``if(...)`` instead of ``if (...)``.
+
+  .. code-block:: cmake
+
+     # Good:
+     if(ENABLE_TESTS)
+       add_subdirectory(tests)
+     endif()
+
+     # Bad:
+     if (ENABLE_TESTS)
+       add_subdirectory(tests)
+     endif()
+
+Commands and Syntax
+-------------------
+
+- **Lowercase Commands**: Always use **lowercase** CMake commands (e.g., ``add_executable``, ``find_package``). This improves readability and consistency.
+
+  .. code-block:: cmake
+
+     # Good:
+     add_library(my_lib STATIC src/my_lib.cpp)
+
+     # Bad:
+     ADD_LIBRARY(my_lib STATIC src/my_lib.cpp)
+
+- **One Command per Line**: Write one CMake command per line to keep the script clean and readable.
+
+  .. code-block:: cmake
+
+     target_sources(my_target PRIVATE
+       src/file1.cpp
+       src/file2.cpp
+     )
+
+Variable Naming
+---------------
+
+- **Use Uppercase for Cache Variables**: When defining cache variables using ``option`` or ``set(... CACHE ...)``, use **UPPERCASE names**.
```

Comment:
Perhaps: **Use Uppercase for Cache Variables or variables shared across CMake files**

as we do have variables that are shared / re-used throughout the CMake tree but are not cache variables.